### PR TITLE
Rely on gradle-baseline provided error-prone

### PR DESCRIPTION
**Pull request by schlosna**
## Before this PR
Error-prone bumps required coordination of build.gradle gradle plugin classpath along with versions.props & gradle-baseline

Closes #1562 https://github.com/palantir/tritium/pull/1543

## After this PR

==COMMIT_MSG==
Rely on gradle-baseline provided error-prone
==COMMIT_MSG==

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -68,6 +67,7 @@ allprojects {
             options.errorprone {
                 option('NullAway:AnnotatedPackages', 'com.palantir')
                 option('NullAway:CheckOptionalEmptiness', 'true')
+                option('AllSuggestionsAsWarnings') // https://github.com/google/error-prone/pull/3301
```
Comment (carterkozak) [inline build.gradle:70]: Very nifty!

Comment (schlosna) [inline build.gradle:70]: This feature isn't yet released, but this should pick it up once we get the bump through gradle-baseline

**Review by carterkozak — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -78,10 +78,6 @@ allprojects {
                         'PreferSafeLoggableExceptions',
                         'PreferSafeLogger'
 
-                // increase strictness for built-in error-prone checks
-                error((com.google.errorprone.scanner.BuiltInCheckerSuppliers.ENABLED_WARNINGS +
-                        com.google.errorprone.scanner.BuiltInCheckerSuppliers.DISABLED_CHECKS
-                        ).collect { it.canonicalName() } as String[])
```
Comment (schlosna) [inline build.gradle:84]: These imports were the main reason for needing `com.google.errorprone:error_prone_core` on gradle classpath. Going to switch over to not-yet-merged or released `AllSuggestionsAsWarnings` from https://github.com/google/error-prone/pull/3301 to opt-in to more strict checks

File: build.gradle (hunk 1 above)
Comment (rickie) [reply build.gradle:70]: Hello, the author of the linked Error Prone ticket here!

I just wanted to point out that we maintain our own [fork](https://github.com/PicnicSupermarket/error-prone) of Error Prone where we already support this feature. 
It's a fork that contains a minimal set of changes on top of mainline Error Prone. We've been using it internally for years.
Just to give you an idea; see the extra commits we have on top of the original `v2.15.0` tag here https://github.com/PicnicSupermarket/error-prone/compare/v2.15.0...v2.15.0-picnic-3.

**Pull request merged**